File: source/cpp/1071. Greatest Common Divisor of Strings.cpp
#include <numeric>
#include <string>
#include <string_view>


class Solution {
public:
    std::string gcdOfStrings(const std::string& str1, const std::string& str2) {
        if (str1 + str2 != str2 + str1) return "";

        return str1.substr(0, std::gcd(str1.size(), str2.size()));
    }
};

File: source/python/1071. Greatest Common Divisor of Strings.py
import math


class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if str1 + str2 != str2 + str1:
            return ""

        return str1[:math.gcd(len(str1), len(str2))]


def test():
    s = Solution()

    assert s.gcdOfStrings("ABCABC", "ABC") == "ABC"
    assert s.gcdOfStrings("ABABAB", "ABAB") == "AB"
    assert s.gcdOfStrings("LEET", "CODE") == ""

    print("All tests passed!")


if __name__ == "__main__":
    test()
